app/components/shared/RoutedTabsHeader.js

```diff
@@ -17,6 +17,11 @@ class RoutedTabsHeader extends React.Component {
 
   componentDidMount() {
     this.updateCaretPosition();
+    window.addEventListener("resize", this.updateCaretPosition);
```

Comment:
I do not like this approach.

We have the habit to add eventListerners to the `App.js` file at containers. Also, many places use the `RoutedTabsHeader` component, and `updateCaretPosition` updates the component state which will make it re-render every resize and this can be costly. 

I noticed the caret updates to a wrong position when the sidebar goes to the bottom. You can add the `sidebarOnBottom` selector and check if it has updated, we have a similar approach at the sidebar.